telemetry/src/lib.rs

```diff
@@ -0,0 +1,235 @@
+use std::{collections::VecDeque, sync, sync::Arc, time::Duration};
+
+use observability_deps::tracing::error;
+use serde::Serialize;
+use tokio::sync::mpsc;
+
+/// This is the core type exposed from this crate which can be
+/// cloned in different places to send the `InternalTelemetryMessage`.
+#[derive(Clone)]
+pub struct TelemetryHandle {
+    /// mpsc sender, safe to clone
+    sender: mpsc::Sender<InternalTelemetryMessage>,
+}
+
+impl TelemetryHandle {
+    pub async fn new(
+        instance_id: String,
+        os: String,
+        influx_version: String,
+        storage_type: String,
+        cores: u32,
+    ) -> Self {
+        let store = TelemetryStore::new(instance_id, os, influx_version, storage_type, cores);
+        // TODO: decide the buffer size
+        let (sender, receiver) = mpsc::channel(100);
+        let telem_msg_handler = TelemetryMesageHandler::new(receiver, store.clone());
+        telem_msg_handler.handle_messages().await;
+        send_telemetry_in_background(store.clone()).await;
```

Comment:
Feels like the store should be Arc wrapped if you're cloning it.

Reply:
mpsc::Sender is a cheap clone because [deep down it uses an Arc](https://docs.rs/tokio/latest/src/tokio/sync/mpsc/chan.rs.html#19-21) and is meant to be cloned. No need to add it here